modin/core/dataframe/pandas/utils.py

```diff
@@ -44,3 +47,34 @@ def concatenate(dfs):
                 i, pandas.Categorical(df.iloc[:, i], categories=union.categories)
             )
     return pandas.concat(dfs)
+
+
+def merge_partitioning(left, right, axis=1):
+    """
+    Get the number of splits across the `axis` for the two dataframes being concatenated.
+
+    Parameters
+    ----------
+    left : PandasDataframe
+    right : PandasDataframe
+    axis : int, default: 1
```

Comment:
> Should we make this parameter as a positional and require it to always be 1? Because we don't perform merge/join along axis=0.

I don't know, why should we limit the functionality if it works perfectly fine with both axes? We might want to use this function somewhere else one day as it serves to merge partitioning for frames that are being generally concatenated, not just merged/joined.

> should we move this function to modin.core.storage_formats.pandas.utils

good point, did so